### PR TITLE
Respect pixel's centers while transform

### PR DESCRIPTION
**Pull request by homm**
This fixes many visual glitches in transforms: shifts, displacements, black borders, extra blurring and other. Just some examples of previous results:

 | Rotating by 0° with `BILINEAR` | Resizing in 1.5 times and back in 0.(6) times with `NEAREST` | Rotating by 90° with `BILINEAR`
---|---------------|--------------|---
Glitches | Blur | Displacements | Blur and black borders
Result:  | _out 0 rotate 0 filter 2 | _out 0 resize 1 5 filter 0 | _out 0 rotate 90 filter 2 
Fixed: | _out 0 rotate 0 filter 2 | _out 0 resize 1 5 filter 0 |_out 0 rotate 90 filter 2

### CODE REVIEW

**Comment by homm** [comment]
Hmm…
```
======================================================================
ERROR: TestFilePcx.test_odd
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/python-pillow/Pillow/Tests/test_file_pcx.py", line 35, in test_odd
    self._roundtrip(hopper(mode).resize((511, 511)))
  File "/home/travis/build/python-pillow/Pillow/Tests/test_file_pcx.py", line 16, in _roundtrip
    self.assert_image_equal(im2, im)
  File "/home/travis/build/python-pillow/Pillow/Tests/helper.py", line 75, in assert_image_equal
    if a.tobytes() != b.tobytes():
  File "PIL/Image.py", line 674, in tobytes
    self.load()
  File "PIL/ImageFile.py", line 240, in load
    raise_ioerror(err_code)
  File "PIL/ImageFile.py", line 59, in raise_ioerror
    raise IOError(message + " when reading image file")
IOError: buffer overrun when reading image file
-------------------- >> begin captured logging << --------------------
```

File: PIL/Image.py [hunk 1]
```diff
@@ -1877,7 +1877,7 @@ def __transformer(self, box, image, method, data,
             xs = float(x1 - x0) / w
             ys = float(y1 - y0) / h
             method = AFFINE
-            data = (xs, 0, x0 + xs/2, 0, ys, y0 + ys/2)
+            data = (xs, 0, x0, 0, ys, y0)
```
Comment (homm) [inline PIL/Image.py]: Looks like `xs/2` was designed to compensate for the `AFFINE` transposition bugs and not required anymore.

File: Tests/test_file_pcx.py [hunk 2]
```diff
@@ -32,6 +32,8 @@ def test_odd(self):
         for mode in ('1', 'L', 'P', 'RGB'):
             # larger, odd sized images are better here to ensure that
             # we handle interrupted scan lines properly.
+            if mode == 'P':
+                continue
```
Comment (homm) [inline Tests/test_file_pcx.py]: Do not forget to remove this.

**Comment by homm** [comment]
Tests was failed due to bug in PCX encoder: https://github.com/python-pillow/Pillow/issues/2023

> Coverage increased (+0.2%) to 78.263%

Yay!

**Comment by homm** [comment]
Merge after #2041

File: Tests/test_file_pcx.py (hunk 2 above)
Comment (wiredfool) [inline Tests/test_file_pcx.py]: Do you want to remove this now?

**Comment by homm** [comment]
@wiredfool done

**Pull request merged**